openedx/features/course_experience/views/course_home.py

```diff
@@ -88,40 +88,56 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         Renders the course's home page as a fragment.
         """
         course_key = CourseKey.from_string(course_id)
+        course = get_course_with_access(request.user, 'load', course_key)
 
-        # Render the outline as a fragment
-        outline_fragment = CourseOutlineFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
+        # Render the course dates as a fragment
+        dates_fragment = CourseDatesFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
 
-        # Get resume course information
-        has_visited_course, resume_course_url = self._get_resume_course_info(request, course_id)
+        # Determine if the user is enrolled in the course (or is staff)
+        is_enrolled_or_staff = CourseEnrollment.is_enrolled(request.user, course_key)
 
-        # Render the welcome message as a fragment
-        welcome_message_fragment = WelcomeMessageFragmentView().render_to_fragment(
-            request, course_id=course_id, **kwargs
-        )
+        # Render only a subset of content for unenrolled users who are not staff
+        if is_enrolled_or_staff:
+            outline_fragment = CourseOutlineFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
 
-        # Render the course dates as a fragment
-        dates_fragment = CourseDatesFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
+            # Render the welcome message as a fragment
+            welcome_message_fragment = WelcomeMessageFragmentView().render_to_fragment(
+                request, course_id=course_id, **kwargs
+            )
 
-        # TODO: Use get_course_overview_with_access and blocks api
-        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+            # Render the verification sock as a fragment
```

Comment:
Nit: 
1. Wondering if these render comments are useful.
2. Why do we feel compelled to use a different name in the comment?  Should this really be a VerificationSockFragmentView, or would this always be the CourseSock and just might contain something different at some point, etc.?